designdocs/batched_messages.md

```diff
@@ -10,98 +10,198 @@
+message and announce the batch location on-chain, instead of announcing each
+individual message.
+
+However, this idea does not go far enough. Batching allows us to support posting
+massive amounts of data, but it would still be expensive to post data of this
+size on chain.
+
+We can leverage off chain storage to make posting large message collections cheap, but we need to
+find a way to constrain the message types posted offchain so that message
+consumers know what types of data to expect. This document aims to explore what
+a system that does the above could look like.
 
 ## Goals and Non-Goals
-This specifies how batches are to be announced on chain; what is required and how a batch may be partially verified based on on-chain information.
+This specifies how messages are to be announced on chain; what is required and
+how a batch may be partially verified based on on-chain information.
+
+This document specifes how messages can be inferred from both schema format type
+and payload location.
 
-This document does not describe the types of DSNP messages that would be in batch announcements. Batch announcements must reference a schema through its ID number. A schema may or may not describe a DSNP message.
+This document also specifies how schemas will constrain the shape of off chain
+messages.
 
-This document does not discuss validation of batch file format located at the URI in the announcement, since
-the file format cannot be verified on-chain. For details about batch files themselves, see [DSNP Spec: Batch Publications](https://spec.dsnp.org/DSNP/BatchPublications).
+This document does not describe the types of DSNP Announcements that will be
+described by schemas. In theory, any message model can be supported.
+
+This document also does not discuss validation of either model or model type. If
+this type of validation is necessary, it should be described elsewhere.
 
 ## Proposal
 * All names are placeholders and may be changed.
 * Types may change as needed during implementation phase
 * Errors in the extrinsic(s) must have different, reasonably-named error enums for each type of error for ease of debugging.
 
-### Constants
-* `BatchPageSizeMax` - the maximum number of batch announcements that will be returned by the `get_batches` query
-* `BatchSizeMinBytes` - the minimum possible size of a valid batch file with 1 row
-
 ### Enums
-* `BatchFormat` - a list of supported formats for batch files. Currently only Parquet file format is supported, however,
-other formats are being considered.
+* `ModelType` - supported serialization formats for message payloads files. Currently only [Parquet](https://parquet.apache.org/docs/) and
+  [Avro](https://avro.apache.org/docs/current/) are supported.
+* `PayloadLocation` - The location of the payload. Can be either `OnChain` or `OffChain`.
+  * `OnChain`
+  * `OffChain`
+* `Payload`
+  * `OnChain`
+    * `source`: `MsaId`
+    * `payload`: `Vec<u8>`
+  * `IPFS`
```

Comment:
any reason why we don't need `source` for IPFS?

Reply:
On-chain data can have this source specified and validated on-chain, but off-chain data, it is not as simple.

That said I wonder if we should shift "on-chain delegation" (aka source enabled/disabled) to a boolean flag (via enum for readability perhaps) on the schema. @dopatraman thoughts?

For those schemas not using batching like DSNP, they might have all that data coming from a delegated source.

It feels complex, but perhaps we actually reduce the complexity overall. First, around payload because then source is "always" there (just sometimes required to be empty by the schema flag for it) and then the payload is simpler. Second, we don't have an AND. The payload location doesn't ALSO define if source is a thing, which isn't really a part of payload location.

Reply:
On additional conversation, If we shift that to a separate flag we can make that a separate story or design doc.